appserver/ejb/ejb-http-remoting/client/src/main/java/fish/payara/ejb/http/client/RemoteEJBContextFactory.java

```diff
@@ -98,12 +116,11 @@ public Context getInitialContext(Hashtable<?, ?> environment) throws NamingExcep
     }
 
     private void updateEnvironmentFromSystemProperties(Hashtable<String, Object> environment) {
-        for (String key : SYSTEM_PROPERTY_KEYS) {
-            if (!environment.containsKey(key)) {
-                String systemPropertyName = key.replaceFirst("fish\\.payara\\.|java\\.naming\\.", "fish.payara.ejb.http.client.");
-                String value = System.getProperty(systemPropertyName, null);
+        for (String environmentVariable : SYSTEM_PROPERTY_KEYS) {
+            if (!environment.containsKey(environmentVariable)) {
```

Comment:
The method name `update...` suggests that you can actually update the environment map from system properties. But this if allows only to set missing values.

Reply:
renamed it.